cg_lims/EPPs/udf/calculate/revio_abc_volumes.py

```diff
@@ -0,0 +1,132 @@
+    annealing_mix_volume: float = artifact.udf[volume_udf]
+    artifact.udf["Volume Annealing Mix (ul)"] = round(annealing_mix_volume, 2)
+    artifact.put()
+
+
+def set_polymerase_dilution_mix_per_sample(
+    artifact: Artifact, volume_udf: str, polymerase_dilution_mix_ratio: str
+) -> None:
+    """Set the volume Polymerase Dilution Mix to add per sample."""
+    polymerase_dilution_volume: float = (
+        float(polymerase_dilution_mix_ratio) * artifact.udf[volume_udf]
+    )
+    artifact.udf["Volume Polymerase Dilution Mix (ul)"] = round(polymerase_dilution_volume, 2)
+    artifact.put()
+
+
+def calculate_and_set_total_ABC_volumes(
+    process: Process,
+    artifacts: List[Artifact],
+    factor: str,
+    volume_udf: str,
+    annealing_reagent_ratio: str,
+    polymerase_buffer_ratio: str,
+    polymerase_dilution_mix_ratio: str,
+    sequencing_polymerase_ratio: str,
+) -> None:
+    """Calculate and set the total master mix and reagent volumes needed for all samples in the step.
+    Adding some excess specified in the cli command."""
+    total_sample_volume: float = calculate_total_sample_volume(
+        artifacts=artifacts, volume_udf=volume_udf
+    )
+
+    process.udf["Total Annealing Mix Volume (ul)"] = round(float(factor) * total_sample_volume, 1)
+    process.udf["Annealing Buffer Volume (ul)"] = round(
+        float(factor) * float(annealing_reagent_ratio) * total_sample_volume, 1
+    )
+    process.udf["Standard Sequencing Primer Volume (ul)"] = round(
+        float(factor) * float(annealing_reagent_ratio) * total_sample_volume, 1
+    )
+    process.udf["Total Polymerase Dilution Mix Volume (ul)"] = round(
+        float(factor) * float(polymerase_dilution_mix_ratio) * total_sample_volume, 1
+    )
+    process.udf["Polymerase Buffer Volume (ul)"] = round(
+        float(factor) * float(polymerase_buffer_ratio) * total_sample_volume, 1
+    )
+    process.udf["Sequencing Polymerase Volume (ul)"] = round(
+        float(factor) * float(sequencing_polymerase_ratio) * total_sample_volume, 1
+    )
+    process.put()
+
+
+@click.command()
+@options.volume_udf()
+@options.factor()
+@options.annealing_reagent_ratio()
+@options.polymerase_buffer_ratio()
+@options.polymerase_dilution_mix_ratio()
+@options.sequencing_polymerase_ratio()
+@click.pass_context
+def revio_abc_volumes(
+    ctx: click.Context,
+    volume_udf: str,
+    factor: str,
+    annealing_reagent_ratio: str,
+    polymerase_buffer_ratio: str,
+    polymerase_dilution_mix_ratio: str,
+    sequencing_polymerase_ratio: str,
+):
+    """Calculate and set the reagent volumes needed for the ABC procedure for SMRTbell libraries."""
+
+    LOG.info(f"Running {ctx.command_path} with params: {ctx.params}")
+
+    process: Process = ctx.obj["process"]
+
+    try:
+        artifacts: List[Artifact] = get_artifacts(process=process)
+
+        for artifact in artifacts:
+            if artifact.udf.get(volume_udf) == None:
```

Comment:
```suggestion
            if artifact.udf.get(volume_udf) is None:
```
This changes nothing functionally, but it is more common to write such statements with 'is'. You could even do something like `if not artifact.udf.get(volume_udf)` if you'd prefer :)